pkgs/development/python-modules/nose/0002-nose-python-3.12-fixes.patch

```diff
@@ -0,0 +1,293 @@
+diff --git a/nose/importer.py b/nose/importer.py
+index e677658..f0cf125 100644
+--- a/nose/importer.py
++++ b/nose/importer.py
+@@ -7,9 +7,10 @@ the builtin importer.
+ import logging
+ import os
+ import sys
++import tokenize
+ from nose.config import Config
+-
+-from imp import find_module, load_module, acquire_lock, release_lock
++from importlib import _imp
++from importlib import machinery
+ 
+ log = logging.getLogger(__name__)
+ 
+@@ -20,6 +21,244 @@ except AttributeError:
+         return (os.path.normcase(os.path.realpath(src)) ==
+                 os.path.normcase(os.path.realpath(dst)))
+ 
++################################################################################
++# BEGIN IMPORTLIB SHIMS
++################################################################################
++
++# Adapted from CPython 3.11.
++# Licensed under the PSLv2.
++# https://github.com/python/cpython/blob/3.11/LICENSE
```

Comment:
This is just a nit because the practice is really common, but:

```
2. Subject to the terms and conditions of this License Agreement, PSF hereby
grants Licensee a nonexclusive, royalty-free, world-wide license to reproduce,
analyze, test, perform and/or display publicly, prepare derivative works,
distribute, and otherwise use Python alone or in any derivative version,
provided, however, that PSF's License Agreement and PSF's notice of copyright,
i.e., "Copyright (c) 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010,
2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022, 2023 Python Software Foundation;
All Rights Reserved" are retained in Python alone or in any derivative version
prepared by Licensee.

3. In the event Licensee prepares a derivative work that is based on
or incorporates Python or any part thereof, and wants to make
the derivative work available to others as provided herein, then
Licensee hereby agrees to include in any such work a brief summary of
the changes made to Python.
```

Would you be willing to make the patch add the full upstream `LICENSE` file as `LICENSE.cpython` file, and adapt the comment similarly to the following?

```
# Adapted from the CPython 3.11 imp.py code.
# Copyright (c) 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010,
2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022, 2023 Python Software Foundation; All Rights Reserved
# Originally licensed under the PSLv2 (see LICENSE.cpython) and incorporated under the LGPL 2.1 (see lgpl.txt).
```

Reply:
Hopefully all is set on the `pynose` side from https://github.com/mdmintz/pynose/issues/33#issuecomment-2221149163. Let me know if anything else!